### ISSUE TITLE
24.11.1 not clearing coverage history between restarts

### ISSUE THREAD

**Issue by CyberiaResurrection** [Zac-HD/hypofuzz#45]
As can be seen in the screenshot below, on Fedora 40, hypofuzz 24.11.1 with python 3.10 and hypothesis 6.122.1 isn't clearing the coverage history on a restart, at least for the case of running exactly one hypothesis test under hypofuzz:

![image](https://github.com/user-attachments/assets/edcf0b7c-3183-4b71-ba3b-79dc9599ea96)

Each of the vertical drops downwards is another hypofuzz restart.

24.9.1 did _not_ have this problem, at least with the python and hypothesis versions listed - it cleared coverage history on each restart.

**Comment by Zac-HD** [comment]
As of PR #41, that's the intended behavior!  We _want_ to persist coverage information etc. between restarts, because this makes our estimators much more accurate and thus we can allocate time between multiple tests more accurately.

On the other hand, we're well aware that this isn't the ideal way to display it on the dashboard.  There are some ideas for how we could improve this in issue #3, but basically we're bottlenecked on someone having time to experiment with analysis of the data we're now collecting, and then implementing that for the dashboard.  This is tricky because restarts are the _simplest_ case we need to handle; there's also the possibility of multiple workers running at the same time, and of the behavior of the system under test changing between runs.

-------------------

So with that context, what would you want to see on this summary page, and/or on the test-specific page?

I can imagine wanting to toggle between latest-run-only (ie old version), all-runs (ie this version), reconstructed-as-if-it-was-all-one-run (not yet implemented), better distinguishing between replaying known inputs and generating new ones...

**Comment by CyberiaResurrection** [comment]
Yeah, I would like to (at least have the option to) restore the pre 24.11 behaviour (displaying latest-run-only) on both the summary and per-test page.

**Comment by tybug** [comment]
I think we could implement a latest-run-only toggle more cleanly with a combination of marks on the x axis and the already existing dragclick-to-subset-graph functionality. The x axis markers (alternatively: colored shading?) could indicate the start of a new run, or phases within that run (replay, generate, distill). Clicking a marker for a run might subset the graph into it. Lots of options to explore here.

**Comment by Zac-HD** [comment]
The [latest version of the dashboard](https://hypofuzz.com/example-dashboard/) is (imo) much improved; we have some pretty good logic for combining reports from multiple runs _without_ counting the "dead time" of replaying covering examples to get back up to the previous state:

![Image](https://github.com/user-attachments/assets/f1b72aba-00b1-4b47-8c9e-4fd8e57b8c09)

I do agree though that it would be good to have a toggle for `[all / latest]`, in addition to our existing inputs/time and linear/log toggles.  On the per-test pages, that should I think be a three-way `[together / separate / latest]`, where together=all the middle option shows an individual line for each worker process which has fuzzed that test.

We've also discussed having a "hide data before [yyyy-mm-dd]" date-based filter, which would make it easy to exclude older reports that you no longer care about.
